### ISSUE TITLE
 Could not find or load main class 

### ISSUE THREAD

**Issue by UditVis** [spring-projects/spring-ide#187]
Everything is configured properly. Using jdk1.8 in build path. While trying to run a basic application as Spring Boot App, Error thrown Could Not Find or Load Main Class. .class files not generated in build path

**Comment by kdvolder** [comment]
It works for me. So not sure why it might not be working for you. 

Let's try figure it out. 

Can you check if "Automatic Workspace Builds" are enabled?

See this SO question / answer which is about turning it off. But you can also switch it back on the same way, or just to check if its enabled or not.

If it is already enabled, we'll need more details to try to reproduce why your project does not seem to be building properly. The easiest is probably if you zip up the project contents and attach it here.

**Comment by Udit2Sep** [comment]
Hey kdvolder,
"Build Automatically" is turned on/checked. Also the project is very basic.
I just went to File -> New -> SpringStarterProject -> Name as 'Demo', Group and Package as 'com.demo', Packaging Jar, Java 1.8, Type Maven,  ->  Next -> Spring Dependencies -> Check Actuator and Web -> Next -> Finish

After that I just try to Run As -> Spring Boot App. It says in console ::: Error: Could not find or load main class com.demo.DemoApplication.

As simple as that. It works perfectly fine on another system. What might be the problem?

Thanks

**Comment by kdvolder** [comment]
> What might be the problem?

To be honest, I'm stumped. Something is screwy but I'm really not sure what it could be.

Some things to try, in order from least drastic to more drastic:

- try selecting the project and refresh it (Right click and choose "Refresh")
- try selecting project and select "Maven >> Update Project" menu
- close STS and reopen it.
- delete the project from workspace (but not the contents on disk) and re-import it into the workspace as 'Existing Maven project').
- open a brand new workspace and import project into the new workspace.
- install a fresh copy of STS and open a new workspace, then try again

**Comment by kdvolder** [comment]
Also take a look in the error log (Window >> Show View >> Other >> Error log) see any errors in there that may give clues on why project doesn't seem to be building.

Also, attaching the exact project contents may be useful, maybe its nothing special, or maybe there's something 'weird' that damaged it. I don't really know, but either way, attaching it really wouldn't hurt.

**Comment by identigral** [comment]
I've experienced this too. I don't have a simple project to repro with but the issue has to do with STS/Eclipse not doing an auto-build when the compiled / derived assets are deleted. In my case, the auto-build is turned on but it's not kicked off after a delete is done via Maven clean from within STS using embedded Maven. I have to do a Maven > Update to force it to build the project. Once that is done, running it as a Boot app works.

**Comment by muld3r** [comment]
Having the same issue. After updating STS to 3.9.1 I'm not able to run any application from Spring Boot Dashboard.

The suggested workarounds above don't resolve the issue for me.

**Comment by kdvolder** [comment]
Folks, I'm going to close this ticket. Its turning into a 'me too' thread but without really seeming to convergen on an actual problem in the IDE we can fix (but more of a collection of various things that can go wrong with workspace settings and maven caches, which aren't exactly 'bugs' we can fix in STS).

It is okay for people to continue to add comments and details and if something that looks 'fixable' emerges we can re-open this or create a more specific new ticket around that. 

Or... the various comments can just sit here, get indexed by google and may help someone stumbling over it with a similar problem.

**Comment by akhilreddyk** [comment]
http://javarevisited.blogspot.sg/2015/04/error-could-not-find-or-load-main-class-helloworld-java.html

**Comment by dhananjayppatankar** [comment]
I tried commentind the test class under com.java.test packages.
It complied and worked for me !!!

**Comment by JanisSharma** [comment]
I got this error because I was trying to run

javac HelloWorld.java && java HelloWorld.class
when I should have removed .class:

javac HelloWorld.java && java HelloWorld

Get Full Code At [Here](https://icetutor.com/question/error-could-not-find-or-load-main-class/)

**Comment by martinlippert** [comment]
@JanisSharma Not sure what exactly you are asking about here. This is the issue tracker for the Spring Tools 4 project. Your question doesn't seem to be related to the IDE at all.

**Comment by RamaiahPillala** [comment]
After updating maven , its working fine

**Comment by nikhil2000** [comment]
after trying all the above possible solutions, nothing worked.Hence happen to delete the .metadata folder in the workspace folder in file explorer and re-importing the project.

**Comment by oiego** [comment]
> > What might be the problem?
> 
> To be honest, I'm stumped. Something is screwy but I'm really not sure what it could be.
> 
> Some things to try, in order from least drastic to more drastic:
> 
> * try selecting the project and refresh it (Right click and choose "Refresh")
> * try selecting project and select "Maven >> Update Project" menu
> * close STS and reopen it.
> * delete the project from workspace (but not the contents on disk) and re-import it into the workspace as 'Existing Maven project').
> * open a brand new workspace and import project into the new workspace.
> * install a fresh copy of STS and open a new workspace, then try again

That worked for me

**Comment by Bharanidharan2K** [comment]
> > What might be the problem?
> 
> To be honest, I'm stumped. Something is screwy but I'm really not sure what it could be.
> 
> Some things to try, in order from least drastic to more drastic:
> 
> * try selecting the project and refresh it (Right click and choose "Refresh")
> * try selecting project and select "Maven >> Update Project" menu
> * close STS and reopen it.
> * delete the project from workspace (but not the contents on disk) and re-import it into the workspace as 'Existing Maven project').
> * open a brand new workspace and import project into the new workspace.
> * install a fresh copy of STS and open a new workspace, then try again


Hey Thanks, Its work for me after updating the maven  "Maven >> Update Project"
